rfcs/0132-meson-builds-nix.md

```diff
@@ -0,0 +1,135 @@
+
+This novel building infrastructure should be able to provide at least feature parity with the current quasi-autotools implementation, albeit in a different user interface.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+Currently @p01arst0rm is writing an implementation from scratch.
+
+Here is a table comparing some expected interactions:
+
+|----------------------|--------------------|------------------------------|
+| Action               | Current            | Meson                        |
+|----------------------|--------------------|------------------------------|
+| Configuring          | `./configure`      | `meson setup build_dir`      |
+| Building             | `make`             | `meson -C build_dir compile` |
+| Installing           | `make install`     | `meson -C build_dir install` |
+| Command-line Options | `--enable-gc=true` | `-Dgc=enabled`               |
+|----------------------|--------------------|------------------------------|
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Some possible drawbacks:
+
+- A new build system would require changes on the code
+  + On the other hand, such changes are likely to improve the code base.
+  
+- A new build system requires the developers become familiarized with it
+  - Specially when this build system uses its own description language.
+  + However, Meson is well documented, and its Python-esque language is easy to grasp.
+
+- A new build system indirectly brings its own dependencies to the Nix project
+  - In particular, the reference implementation of Meson is written in Python.
+  - Further, this reference implementation generates script files meant to be consumed by Ninja.
+  - Ninja is a tool written in C++ that acts like a `make` replacement.
+  - This particular setting brings concerns about complexifying the bootstrap route.
+  + Given that Nix is currently written in C++, we can assume a C++ compiler as part of such a bootstrap route.
+  + There are full-featured alternative tools that replace Meson and Ninja. 
+    + Namely, Muon and Samurai are implementations of Meson and Ninja that require only a C compiler and a set of POSIX standard tools.
+  + Autotools also have its own set of dependencies, and a fair comparison should include them
+
+- A new build system would require new strategies from the end users
+  - In particular, package managers that deploy Nix for their respective platforms.
+  + However, Meson and Ninja are a widespread toolset.
+    + Many open source projects use Meson, from mpv and dosbox-staging to Xorg and GNOME
+    + According to Repology, Meson is present in 53 package manager's families
+
+- The transition between between the old and new build systems should be smooth
+  + Meson is not an obscure project; a careful documentation update should be sufficient
```

Comment:
```suggestion
  + The majority of a Meson buildsystem has been developed & tested, it would only need to be implemented with minor additions.
  + Meson is not an obscure project; a careful documentation update should be sufficient
```

Reply:
I did not understand this,

Reply:
just that a lot of the work has been done, so implementing it shouldnt be too difficult